src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -84,6 +84,31 @@ public boolean isSameAppointment(Appointment otherAppointment) {
                 && otherAppointment.timePeriod.equals(timePeriod);
     }
 
+    /**
+     * Returns true if both appointments has overlapping time period.
+     * Overlapping is defined as within, the same or partly coinciding timePeriods.
+     */
+    public boolean hasOverlappingTimePeriod(Appointment otherAppt) {
+        Time existingStartTime = this.getStartTime();
+        Time existingEndTime = this.getEndTime();
+        Time otherStartTime = otherAppt.getStartTime();
+        Time otherEndTime = otherAppt.getEndTime();
+
+        // other startTime is before existing startTime &&
+        // other endTime is before or equal to existing startTime --> return false
+        if (otherStartTime.compareTo(existingStartTime) < 0 && otherEndTime.compareTo(existingStartTime) <= 0) {
+            return false;
+        }
+
+        // other startTime is after or equal to existing endTime
+        // && other endTime is after existing endTime --> return false
+        if (otherStartTime.compareTo(existingEndTime) >= 0 && otherEndTime.compareTo(existingEndTime) > 0) {
```

Comment:
Just wondering, since we already ensure that end time > start time, is the second check necessary?

Reply:
True, may not be necessary. Was thinking it's a double check, in case we change the logic of Time constraints. Should I remove it?